src/lib/index.ts

```diff
@@ -103,10 +103,11 @@ export class GoogleToken {
     };
 
     if (!this.hasExpired()) {
-      return callback(null, this.token);
+      process.nextTick(callback, null, this.token);
+      return;
     } else {
       if (!this.key && !this.keyFile) {
-        callback(new Error('No key or keyFile set.'));
+        process.nextTick(callback, new Error('No key or keyFile set.'));
```

Comment:
Prefer `setImmediate` over `process.nextTick`; through out the PR.

Reply:
context: https://gist.github.com/brycebaril/ff86eeb90b53fd0c523e

Reply:
Curious, why?